baybe/utils/dataframe.py

```diff
@@ -361,6 +361,55 @@ def df_uncorrelated_features(
     return data
 
 
+def df_add_noise_to_degenerate_rows(
+    df: pd.DataFrame, noise_level: float = 0.001
+) -> pd.DataFrame:
+    """Add noise to degenerate rows to make them numerically distinguishable.
+
+    Note that the dataframe is changed in-place and also returned. The dataframe is
+    left untouched if no rows are degenerate.
+
+    Args:
+        df: The dataframe to be modified.
+        noise_level: The magnitude of noise relative to the min-max range of
+            values for each column.
+
+    Returns:
+        The modified dataframe.
+
+    Raises:
+        TypeError: If the provided dataframe has non-numerical content.
+    """
+    # Find degenerate rows, exit if there are none
+    degen_rows = df.duplicated(keep=False)
+    if degen_rows.sum() == 0:
```

Comment:
Really minor but still wanted to point this out, also to potentially raise awareness: `sum` requires iterating through the entire array, while `any` can leverage short-cut Boolean evaluation

Reply:
changed